backend/src/middleware/validation.ts

```diff
@@ -30,11 +41,51 @@ const validateRequest: RequestHandler = (req, res, next) => {
   throw createHttpError(400, errorString);
 };
 
+const validateUser: RequestHandler = asyncHandler(async (req, res, next) => {
+  const authHeader = req.headers.authorization;
+  let token;
+
+  if (authHeader && authHeader.split(" ")[0] === "Bearer") {
+    token = authHeader.split(" ")[1];
+  } else {
+    throw createHttpError(401, "Authorization Token Required");
+  }
+
+  try {
+    const decodedToken = await firebaseAdmin.firebaseAuth.verifyIdToken(token);
+    const email = decodedToken.email;
+
+    const user = await UserModel.findOne({ email });
+    /* eslint-disable-next-line @typescript-eslint/no-unsafe-member-access */
+    req.body.currentUser = user;
+  } catch (error) {
+    throw createHttpError(404, "Error finding user");
+  }
+
+  next();
+});
+
+const validateHousingLocator: RequestHandler = (req, res, next) => {
+  validateUser(req, res, () => {
+    /* eslint-disable-next-line @typescript-eslint/no-unsafe-member-access */
+    if (req.body.currentUser.isHousingLocator) {
+      next();
+    } else {
+      res.status(401).send({ message: "Housing Locators only " });
+    }
+  });
+};
+
 /**
  * Validates a request using express-validator (short-circuiting the handler chain
  * if necessary and returning the error messages to the client).
  *
  * @param validators a list of validation chains to apply to the request
  * @returns the validation chain with the validation handler appended
  */
-export const validateWith = (validators: ValidationChain[]) => [...validators, validateRequest];
+export const validateWith = (validators: ValidationChain[]) => [
+  ...validators,
+  validateRequest,
+  validateUser,
+];
+export { validateHousingLocator };
```

Comment:
I don't think we should include `validateUser` here since this middleware is supposed to be for validating the actual content of requests (e.g. body, params, etc). Also, not all of our backend routes require a user token (like the create Unit route).

```suggestion
export const validateWith = (validators: ValidationChain[]) => [
  ...validators,
  validateRequest,
];
export { validateUser, validateHousingLocator };
```